rustls/src/conn/unbuffered.rs

```diff
@@ -6,19 +6,51 @@ use core::{fmt, mem};
 use std::error::Error as StdError;
 
 use super::UnbufferedConnectionCommon;
+use crate::client::ClientConnectionData;
 use crate::msgs::deframer::DeframerSliceBuffer;
+use crate::server::ServerConnectionData;
 use crate::Error;
 
-impl<Data> UnbufferedConnectionCommon<Data> {
+impl UnbufferedConnectionCommon<ClientConnectionData> {
+    /// Processes the TLS records in `incoming_tls` buffer until a new [`UnbufferedStatus`] is
+    /// reached.
+    pub fn process_tls_records<'c, 'i>(
+        &'c mut self,
+        incoming_tls: &'i mut [u8],
+    ) -> Result<UnbufferedStatus<'c, 'i, ClientConnectionData>, Error> {
+        self.process_tls_records_common(incoming_tls, |_| None::<()>, |_, _, _| unreachable!())
```

Comment:
Does this really need the turbofish?

Reply:
yes, but we can set the closure's type instead which is shorter